pkg/sql/distsqlrun/countrows.go

```diff
@@ -93,10 +93,15 @@ func (ag *countAggregator) Next() (sqlbase.EncDatumRow, *ProducerMetadata) {
 			return nil, meta
 		}
 		if row == nil {
-			ag.MoveToDraining(nil /* err */)
 			ret := make(sqlbase.EncDatumRow, 1)
 			ret[0] = sqlbase.EncDatum{Datum: tree.NewDInt(tree.DInt(ag.count))}
-			return ag.ProcessRowHelper(ret), nil
+			rendered := ag.ProcessRowHelper(ret)
```

Comment:
I wonder if you can call `pb.out.ProcessRow` directly here to not have to worry about `ProcessRowHelper`'s side effects.

Reply:
Or maybe have a non-draining implementation that returns a bool that you use both here and in `ProcessRowHelper`.

Reply:
Done.

Reply:
I would add an `|| err != nil` just to be safe.

Reply:
Fixed this part - realized that it's not even conditional. We always drain here, the only question is whether err is not nil.